### PR TITLE
Migrate docker image to jdk with JEP 358

### LINKED ISSUE #322: JDK 14 with JEP 358: Helpful NullPointerExceptions enabled for docker image
Since NullPointer Exceptions are common in artipie, I would like to suggest to switch to JDK 14 and enable JEP 358.

### PR DESCRIPTION
**Pull request by Sammers21**
In this PR:
- docker container is based on JDK 14
- enabled JEP 358 with Helpful NullPointerExceptions
- introduced variable JVM_OPTS variable, allowing users to pass various JVM flags for tuning

Closes #322.

### CODE REVIEW

**Comment by 0crat** [comment]
@Sammers21 this pull request is too small, just 5 lines changed (less than 10), there will be no formal code review, see [§53](http://www.zerocracy.com/policy.html#53) and [§28](http://www.zerocracy.com/policy.html#28); in the future, try to make sure your pull requests are not too small; @g4s8[/z](https://www.0crat.com/u/g4s8) please review this and merge or reject

File: Dockerfile [hunk 1]
```diff
@@ -6,12 +6,13 @@ COPY src ./src
...
+FROM adoptopenjdk/openjdk14:alpine-jre
+ENV JAVA_OPTS=""
 LABEL description="Artipie binary repository managment tool"
 LABEL maintainer="titantins@gmail.com"
 LABEL maintainer="g4s8.public@gmail.com"
 COPY --from=build_jar /jar/target/artipie-jar-with-dependencies.jar /usr/lib/artipie.jar
 COPY _config.yml /etc/artipie.yml
 EXPOSE 80
-ENTRYPOINT ["java", "-jar", "/usr/lib/artipie.jar"]
+ENTRYPOINT java $JVM_OPTS --enable-preview -XX:+ShowCodeDetailsInExceptionMessages -jar /usr/lib/artipie.jar
```
Comment (g4s8) [inline Dockerfile:17]: @Sammers21 you've introduced env variable called `JAVA_OPTS` but uses `JVM_OPTS`

File: Dockerfile [hunk 2]
```diff
@@ -6,12 +6,12 @@ COPY src ./src
...
+ENV JAVA_OPTS=""
 LABEL description="Artipie binary repository managment tool"
 LABEL maintainer="titantins@gmail.com"
 LABEL maintainer="g4s8.public@gmail.com"
 COPY --from=build_jar /jar/target/artipie-jar-with-dependencies.jar /usr/lib/artipie.jar
 COPY _config.yml /etc/artipie.yml
 EXPOSE 80
-ENTRYPOINT ["java", "-jar", "/usr/lib/artipie.jar"]
-CMD ["--config-file=/etc/artipie.yml", "--port=80"]
+CMD java $JVM_OPTS --enable-preview -XX:+ShowCodeDetailsInExceptionMessages -jar /usr/lib/artipie.jar --config-file=/etc/artipie.yml --port=80
```
Comment (Sammers21) [inline Dockerfile:17]: I use shell form because there is no different way to pass JVM_OPTS.

**Comment by Sammers21** [comment]
@g4s8, can we merge?

**Comment by Sammers21** [comment]
accidentally closed/reopened

**Comment by 0crat** [comment]
@Sammers21 this pull request is too small, just 6 lines changed (less than 10), there will be no formal code review, see [§53](http://www.zerocracy.com/policy.html#53) and [§28](http://www.zerocracy.com/policy.html#28); in the future, try to make sure your pull requests are not too small; @g4s8[/z](https://www.0crat.com/u/g4s8) please review this and merge or reject

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#323` is not assigned, can't get performer

**Comment by Sammers21** [comment]
@g4s8, can we merge this PR?

File: Dockerfile (hunk 2 above)
Comment (g4s8) [reply Dockerfile:17]: @Sammers21 I know that. But the problem is still not fixed: you've introduced env-variable `JAVA_OPTS`, but uses `JVM_OPTS` (different name)

Comment (Sammers21) [reply Dockerfile:17]: @g4s8, just fixed in https://github.com/artipie/artipie/pull/323/commits/8ee88d3c1fd645e71abd10bc957c6f98ad655f98

**Comment by Sammers21** [comment]
@g4s8, can we merge this PR?

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#323` is not assigned, can't get performer

**Pull request merged**